testbed/eks/lib/addons/fluxv2.ts

```diff
@@ -0,0 +1,110 @@
+import cdk = require('@aws-cdk/core');
+import eks = require('@aws-cdk/aws-eks');
+import * as yaml from 'js-yaml';
+import * as request from 'sync-request';
+
+export interface FluxV2Props extends cdk.StackProps {
+    cluster: eks.Cluster;
+    fluxRepos: Repository[];
+    secretName: string;
+    fluxVersion: string;
+}
+export class Repository {
+    repoUrl: string;
+    repoBranch: string;
+    repoPath: string;
+
+    constructor(repoUrl: string, repoBranch?: string, repoPath?: string) {
+        if (!repoBranch) {
+            this.repoBranch = "main"
+        } else {
+            this.repoBranch = repoBranch
+        }
+        if (!repoPath) {
+            this.repoPath = "test/workflows"
+        } else {
+            this.repoPath = repoPath
+        }
+        this.repoUrl = repoUrl
+    }
+}
+class FluxRelease {
+    private fluxVersion: string;
+    private manifestUrl: string;
+    public installManifest: any;
+
+    constructor(version: string) {
+        this.fluxVersion = version;
+        this.manifestUrl = `https://github.com/fluxcd/flux2/releases/download/${this.fluxVersion}/install.yaml`
+    }
+
+    public getUrl(): string {
+        return this.manifestUrl;
+    }
+
+    public getManifest(): any {
+        this.installManifest = yaml.loadAll(
+            request.default('GET', this.manifestUrl)
+                .getBody()
+                .toString()
+        );
+        return this.installManifest;
+    }
+}
+
+export class FluxV2 extends cdk.Construct {
+    constructor(scope: cdk.Construct, id: string, props: FluxV2Props) {
+        super(scope, id);
+
+        // Actually install Flux components onto the cluster
+        const fluxRelease = new FluxRelease(props.fluxVersion);
+        const fluxManifest = props.cluster.addManifest('fluxManifest', ...fluxRelease.getManifest());
+
+        for (let index = 0; index < props.fluxRepos.length; index++) {
```

Comment:
A couple ways to clean this up. Add a function to this class like `addManifest()`. I like the parity here w/ cluster.addManifest as well.

Then in your constructor code `props.repositories.foreach(this.addManifest)`. This will keep your constructor well encapsulated. Typically, if you're finding yourself writing big for loops, it's a good hint that its time to break the logic apart.

Reply:
I don't think encapsulating in `addManifest` function adds any value here as it's only served once.